gstcefnsapplication.h

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: 2024 L. E. Segovia <amy@centricular.com>
+// SPDX-License-Ref: LGPL-2.1-or-later
+
+#pragma once
+
+#import <CoreFoundation/CoreFoundation.h>
+#include <objc/objc.h>
+
+/// Schedules a job to handle CEF's work for immediate execution
+/// on the main thread. The function does not block.
+void gst_cef_loop();
+/// Schedules a timer with the given timeout to trigger handling the next
+/// slice of work. The handling will take place in the main thread.
+/// The function does not block.
+CFRunLoopTimerRef gst_cef_domessagework(CFTimeInterval interval);
+/// Hook on different lifecycle events to trigger CefShutdown.
+/// These are:
+/// - NSApplicationWillTerminateNotification
+/// - NSEvent.NSEventMaskApplicationDefined as sent by gstmacos.mm
+/// The shutdown executes only once.
+void gst_cef_set_shutdown_observer();
+/// Execute [NSThread isMainThread] from a non-Objective-C context.
```

Comment:
Could also call `pthread_main_np()` from C but I guess it doesn't really matter

Reply:
It's perfectly OK, one less function in the Objective-C bridge 😁